app/Jobs/ProcessNewVideo.php

```diff
@@ -143,17 +143,30 @@ public function handleFile($file, $path)
         }
         $this->video->save();
 
-        $times = $this->getThumbnailTimes($this->video->duration);
         $disk = Storage::disk(config('videos.thumbnail_storage_disk'));
         $fragment = fragment_uuid_path($this->video->uuid);
-        $format = config('thumbnails.format');
-        $width = config('thumbnails.width');
-        $height = config('thumbnails.height');
-
         try {
-            foreach ($times as $index => $time) {
-                $buffer = $this->generateVideoThumbnail($path, $time, $width, $height, $format);
-                $disk->put("{$fragment}/{$index}.{$format}", $buffer);
+            $tmp = config('videos.tmp_dir');
+            $tmpDir = "{$tmp}/{$fragment}";
+
+            // Directory for extracted images
+            if (!File::exists($tmpDir)) {
+                File::makeDirectory($tmpDir, 0755, true);
+            }
+
+            if (!$disk->exists($fragment)) {
+                $disk->makeDirectory($fragment);
+            }
```

Comment:
This should not be required. The directories should be automatically created with `put()`.

```suggestion
```